batcher/batcher.go

```diff
@@ -18,15 +18,38 @@ package batcher
 
 import (
 	"errors"
-	"sync"
-	"sync/atomic"
 	"time"
 )
 
-const (
-	batcherActive   = uint32(0)
-	batcherDisposed = uint32(1)
-)
+// I honestly can't believe I'm doing this, but go's sync package doesn't
+// have a TryLock function.
+// Could probably do this with atomics
+type mutex struct {
```

Comment:
Might suggest naming this `tryableMutex` or something else that describes the purpose. `mutex` makes me think `sync.Mutex`.